graphjet-core/src/main/java/com/twitter/graphjet/algorithms/socialproof/EntitySocialProofGenerator.java

```diff
@@ -0,0 +1,178 @@
+  private final Int2ObjectMap<Byte2ObjectMap<Long2ObjectMap<LongSet>>> socialProofs;
+  // Entity (Int) -> Sum of social proof edges (Double)
+  private final Int2DoubleMap socialProofWeights;
+  protected RecommendationType recommendationType;
+  protected IDMask idMask;
+
+  public EntitySocialProofGenerator(
+      NodeMetadataLeftIndexedPowerLawMultiSegmentBipartiteGraph graph,
+      TweetIDMask mask,
+      RecommendationType recommendationType
+  ) {
+    this.graph = graph;
+    this.idMask = mask;
+    this.recommendationType = recommendationType;
+    // Variables socialProofs and socialProofWeights are re-used for each request.
+    this.socialProofs = new Int2ObjectOpenHashMap<>();
+    this.socialProofWeights = new Int2DoubleOpenHashMap();
+  }
+
+  private void updateSocialProofWeight(int entity, double weight) {
+    // We sum the weights of incoming leftNodes as the weight of the rightNode.
+    socialProofWeights.put(
+        entity,
+        weight + socialProofWeights.get(entity)
+    );
+  }
+
+  private void addSocialProof(int entity, byte edgeType, long leftNode, long rightNode) {
+    if (!socialProofs.containsKey(entity)) {
+      socialProofs.put(entity, new Byte2ObjectArrayMap<>());
+      socialProofWeights.put(entity, 0);
+    }
+    Byte2ObjectMap<Long2ObjectMap<LongSet>> socialProofMap = socialProofs.get(entity);
+
+    // Get the user to tweets map variable by the engagement type.
+    if (!socialProofMap.containsKey(edgeType)) {
+      socialProofMap.put(edgeType, new Long2ObjectArrayMap<>());
+    }
+    Long2ObjectMap<LongSet> userToTweetsMap = socialProofMap.get(edgeType);
+
+    // Add the connecting user to the user map.
+    if (!userToTweetsMap.containsKey(leftNode)) {
+      userToTweetsMap.put(leftNode, new LongArraySet());
+    }
+    LongSet connectingTweets = userToTweetsMap.get(leftNode);
+
+    // Add the connecting tweet to the tweet set.
+    if (!connectingTweets.contains(rightNode)) {
+      connectingTweets.add(rightNode);
+    }
+  }
+
+  /**
+   * Collect social proofs for a given {@link SocialProofRequest}.
+   *
+   * @param request contains a set of input ids and a set of seed users.
+   */
+  private void collectRecommendations(EntitySocialProofRequest request) {
+    socialProofs.clear();
+    socialProofWeights.clear();
+    IntSet inputEntityIds = request.getEntityIds();
+    ByteSet socialProofTypes = new ByteArraySet(request.getSocialProofTypes());
+    byte entityType = (byte) recommendationType.getValue();
+
+    // Iterate through the set of seed users with weights. For each seed user, we go through his edges.
+    for (Long2DoubleMap.Entry entry: request.getLeftSeedNodesWithWeight().long2DoubleEntrySet()) {
+      long leftNode = entry.getLongKey();
+      double weight = entry.getDoubleValue();
+      NodeMetadataMultiSegmentIterator edgeIterator =
+        (NodeMetadataMultiSegmentIterator) graph.getLeftNodeEdges(leftNode);
+      if (edgeIterator == null) { continue; }
```

Comment:
style nit: 
either
`if (edgeIterator == null) continue;`
or place the { continue; } on separate lines